src/datasource.ts

```diff
@@ -577,11 +590,33 @@ async function findAssetTimeseries(
     : {
         assetIds: [assetId],
       };
-
   // since /dataquery can only have 100 items and checkboxes become difficult to use past 100 items,
   //  we only get the first 100 timeseries, and show a warning if there are too many timeseries
   const limit = 101;
-  const ts = await getTimeseries({ filter, limit }, connector);
+  let ts = [];
+  if (assetQuery.includeSubTimeseries !== false) {
+    const timeseries = await getTimeseries({ filter, limit }, connector);
+    if (!isEmpty(timeseries)) {
+      ts = concat(ts, ...timeseries);
+    }
+  }
+  if (assetQuery.withRelationships) {
+    const { labels, dataSetIds, sourceExternalIds, limit } = assetQuery.relationshipsQuery;
+    const relationshipsList = await fetchRelationships(
+      {
+        ...filterLabels(labels),
+        ...filterdataSetIds(dataSetIds),
+        ...filterExternalId(sourceExternalIds),
+        ...timeFrame,
+      },
+      limit,
+      connector
+    );
+    if (!isEmpty(relationshipsList)) {
+      ts = concat(ts, ...map(relationshipsList, 'target'));
+    }
+  }
+  ts = uniqBy(ts, 'id');
   if (ts.length === limit) {
     emitEvent(responseWarningEvent, { refId, warning: TIMESERIES_LIMIT_WARNING });
```

Comment:
this limit should be checked on both arrays before you merge them. we need to find a good way of doing that

Reply:
should we emit warning for each timeseries limit warning and also have the combined limit warning?

Reply:
what if the length is bigger then limit? because right now we check only if equals..

Reply:
sure, can be `>=` instead